src/FunderMaps.Core/Entities/Building.cs

```diff
@@ -37,27 +38,33 @@ public Building()
         public bool IsActive { get; set; } = true;
 
         /// <summary>
-        ///     Address identifier.
+        ///     Geometry.
         /// </summary>
         [Required]
-        public string Address { get; set; }
+        public string Geometry { get; set; }
 
         // TODO: Type, see #211
         /// <summary>
         ///     External data source id.
         /// </summary>
-        [Required]
+        [Required(AllowEmptyStrings = false)]
         public string ExternalId { get; set; }
 
         /// <summary>
         ///     External data source.
         /// </summary>
         [Required]
-        public string ExternalSource { get; set; }
+        public ExternalDataSource ExternalSource { get; set; }
+
+        /// <summary>
+        ///     Building type.
+        /// </summary>
+        public BuildingType? buildingType { get; set; }
```

Comment:
Building type hasn't been fully implemented in the database, the format might change later. This is fine for now, just adding this as a note.